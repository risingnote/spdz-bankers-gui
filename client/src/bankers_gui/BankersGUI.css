.BankersGUI-main {
  display: -webkit-flex;
  display: flex;
  flex-flow: row wrap;
  padding: 20px;
}

.BankersGUI-container {
  flex: 1;
}

.BankersGUI-connection {
  flex: 0 1 auto;
}
