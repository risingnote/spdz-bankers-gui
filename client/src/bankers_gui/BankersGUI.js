/**
 * High level layout for the bankers GUI.
 * Responsible for coordinating SPDZ proxy status change with Connection component.
 */
import React, { PropTypes, Component } from 'react'
import { List } from 'immutable'
import { Connection } from 'spdz-gui-components'

import BankersFormContainer from './BankersFormContainer'
import BankersTable from './BankersTable'

import './BankersGUI.css'

class BankersGUI extends Component {
  constructor (props) {
    super(props)
    this.state = {
      spdzProxyStatus : [],
      dinersList: [],
      winningClientId: undefined
    }
    this.updateProxyConnectionStatus = this.updateProxyConnectionStatus.bind(this)
    this.changeToDiners = this.changeToDiners.bind(this)  
  }

  /**
   * Update the proxy connection status for each spdz proxy.
   * @param proxyStatus array of objects {id: index of spdzProxyServerList, status: ProxyStatusCodes}
   */
  updateProxyConnectionStatus(proxyStatus) {
    this.setState({spdzProxyStatus: proxyStatus})
  }

  changeToDiners(dinersList, winningClientId) {
    this.setState({dinersList: dinersList, winningClientId: winningClientId})
  }

  render() {
    const connectionHeaderStyle = {
        color: 'rgb(28, 118, 152)',
        fontWeight: 'bold'
    }    
    return (
      <div className="BankersGUI-main">
        <div className="BankersGUI-component">
          <BankersFormContainer className="BankersGUI-component"
                                updateConnectionStatus={this.updateProxyConnectionStatus}
                                changeToDiners={this.changeToDiners}
                                spdzProxyServerList={this.props.spdzProxyServerList.toArray().map(immutableMap => immutableMap.toObject())}
                                spdzApiRoot={this.props.spdzApiRoot}
                                clientPublicKey={this.props.clientPublicKey} />
        </div>
        <div className="BankersGUI-component">
          <BankersTable className="BankersGUI-component"
                        diners={this.state.dinersList}
                        winningClientId={this.state.winningClientId} />
        </div>
        <div className="BankersGUI-component">
          <Connection className="BankersGUI-component"
                      spdzProxyServerList={this.props.spdzProxyServerList}
                      spdzProxyStatus={this.state.spdzProxyStatus}
                      headerStyle={connectionHeaderStyle} />
        </div>
      </div>
    )
  }
}

BankersGUI.propTypes = {
  spdzProxyServerList: PropTypes.instanceOf(List).isRequired,
  spdzApiRoot: PropTypes.string.isRequired,
  clientPublicKey: PropTypes.string.isRequired
}

export default BankersGUI
