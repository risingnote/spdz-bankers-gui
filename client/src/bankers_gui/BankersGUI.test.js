import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { List, Map } from 'immutable'

import BankersGUI from './BankersGUI'

const spdzProxyConfig = List.of(
      Map({
        url: "http://spdzProxy.one:4000",
        publicKey: "0102030405060708010203040506070801020304050607080102030405060708", 
        encryptionKey: "AA02030405060708010203040506070801020304050607080102030405060708"
      }),
      Map({
        url: "http://spdzProxy.two:4000",
        publicKey: "3302030405060708010203040506070801020304050607080102030405060708",         
        encryptionKey: "BB02030405060708010203040506070801020304050607080102030405060708"
      })
    )

describe('Bankers GUI display component rendering', () => {
  it('Renders as expected (compared to a snapshot).', () => {

    const tree = renderer.create(
      <BankersGUI spdzProxyServerList={spdzProxyConfig} spdzApiRoot={'/spdzapi'} clientPublicKey={'a1b2c3d4'}/>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Checks proxy connection status state is set after component mounts', () => {
    
    const wrapper = mount(<BankersGUI spdzProxyServerList={spdzProxyConfig} 
                                      spdzApiRoot={'/spdzapi'} clientPublicKey={'a1b2c3d4'}/>)

    expect(wrapper.state().spdzProxyStatus).toEqual([])
  })

})
