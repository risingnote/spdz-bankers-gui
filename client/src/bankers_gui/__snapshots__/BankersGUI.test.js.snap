exports[`Bankers GUI display component rendering Renders as expected (compared to a snapshot). 1`] = `
<div
  className="BankersGUI-main">
  <div
    className="BankersGUI-container">
    <div
      className="Bankers">
      <form
        className="BankersForm"
        onSubmit={[Function]}>
        <h4>
          Bankers Celebration Dinner
        </h4>
        <p>
          ...but who should pay?
        </p>
        <label
          htmlFor="joinName">
          Join meal as
        </label>
        <input
          disabled=""
          id="joinName"
          onChange={[Function]}
          type="text"
          value="" />
        <label
          htmlFor="bonusValue">
          Bonus
        </label>
        <input
          disabled=""
          id="bonusValue"
          onChange={[Function]}
          type="text"
          value="" />
        <input
          disabled=""
          type="submit"
          value="Send" />
        <p
          className="smallText">
          
        </p>
      </form>
      <div
        className="BankersTable">
        <div
          className="place"
          style={
            Object {
              "transform": "translate(0rem, -8.2rem) rotate(180deg)",
            }
          }>
          <div
            className="place-setting">
            <div
              style={
                Object {
                  "transform": "rotate(-180deg)",
                }
              }>
              
            </div>
          </div>
        </div>
        <div
          className="place"
          style={
            Object {
              "transform": "translate(5.8rem, -5.8rem) rotate(225deg)",
            }
          }>
          <div
            className="place-setting">
            <div
              style={
                Object {
                  "transform": "rotate(-225deg)",
                }
              }>
              
            </div>
          </div>
        </div>
        <div
          className="place"
          style={
            Object {
              "transform": "translate(8.2rem, 0rem) rotate(270deg)",
            }
          }>
          <div
            className="place-setting">
            <div
              style={
                Object {
                  "transform": "rotate(-270deg)",
                }
              }>
              
            </div>
          </div>
        </div>
        <div
          className="place"
          style={
            Object {
              "transform": "translate(5.8rem, 5.8rem) rotate(315deg)",
            }
          }>
          <div
            className="place-setting">
            <div
              style={
                Object {
                  "transform": "rotate(-315deg)",
                }
              }>
              
            </div>
          </div>
        </div>
        <div
          className="place"
          style={
            Object {
              "transform": "translate(0rem, 8.2rem) rotate(0deg)",
            }
          }>
          <div
            className="place-setting">
            <div
              style={
                Object {
                  "transform": "rotate(-0deg)",
                }
              }>
              
            </div>
          </div>
        </div>
        <div
          className="place"
          style={
            Object {
              "transform": "translate(-5.8rem, 5.8rem) rotate(45deg)",
            }
          }>
          <div
            className="place-setting">
            <div
              style={
                Object {
                  "transform": "rotate(-45deg)",
                }
              }>
              
            </div>
          </div>
        </div>
        <div
          className="place"
          style={
            Object {
              "transform": "translate(-8.2rem, 0rem) rotate(90deg)",
            }
          }>
          <div
            className="place-setting">
            <div
              style={
                Object {
                  "transform": "rotate(-90deg)",
                }
              }>
              
            </div>
          </div>
        </div>
        <div
          className="place"
          style={
            Object {
              "transform": "translate(-5.8rem, -5.8rem) rotate(135deg)",
            }
          }>
          <div
            className="place-setting">
            <div
              style={
                Object {
                  "transform": "rotate(-135deg)",
                }
              }>
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="s-alert-wrapper" />
    </div>
  </div>
  <div
    className="BankersGUI-connection">
    <div
      className="Connection-panel">
      <p
        className="Connection-subHead">
        Spdz Proxy Servers
      </p>
      <div
        className="ProxyStatus-row">
        <div
          className="ProxyStatus-col1 ProxyStatus-marker-disconnected" />
        <div
          className="ProxyStatus-col2 smallText">
          http://spdzProxy.one:4000
        </div>
      </div>
      <div
        className="ProxyStatus-row">
        <div
          className="ProxyStatus-col1 ProxyStatus-marker-disconnected" />
        <div
          className="ProxyStatus-col2 smallText">
          http://spdzProxy.two:4000
        </div>
      </div>
    </div>
  </div>
</div>
`;
